src/mscorlib/src/System/Text/StringBuilder.cs

```diff
@@ -999,6 +999,124 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             return this;
         }
 
+        public StringBuilder Append(StringBuilder value)
+        {
+            Contract.Ensures(Contract.Result<StringBuilder>() != null);
+
+            if (value != null && value.Length > 0)
+            {
+                AppendCore(value, 0, value.Length);
+            }
+            return this;
+        }
+
+        public StringBuilder Append(StringBuilder value, int startIndex, int count)
+        {
+            if (startIndex < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(startIndex), Environment.GetResourceString("ArgumentOutOfRange_GenericPositive"));
+            }
+            if (count < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count), Environment.GetResourceString("ArgumentOutOfRange_GenericPositive"));
+            }
+            Contract.Ensures(Contract.Result<StringBuilder>() != null);
+            Contract.EndContractBlock();
+
+            if (value == null)
+            {
+                if (startIndex == 0 && count == 0)
+                {
+                    return this;
+                }
+                throw new ArgumentNullException(nameof(value));
+            }
+            if (count > value.Length - startIndex)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count), Environment.GetResourceString("ArgumentOutOfRange_IndexLength"));
+            }
+
+            if (count > 0)
+            {
+                AppendCore(value, startIndex, count);
+            }
+            return this;
+        }
+
+        private unsafe void AppendCore(StringBuilder value, int startIndex, int count)
```

Comment:
StringBuilder is not thread safe so any its method have such racing. ToString also not thread safe. So we can write not thread save Append method to append. I do not know who create race comment in ToString methods but it protect from nothing because it not thread safe.
I sagest just protect from append it self into it self. Now new method will create not correct result, but old append method that use ToString return correct result. So I will add Append to it self protection by using ToString before Append to its self.